### PR TITLE
Add support for Stable Diffusion

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR adds optimized support for Stable Diffusion inference on Habana Gaudi.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,6 +172,39 @@ gaudi_config = GaudiConfig.from_pretrained(
...
+
+- scheduler = DDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
++ scheduler = GaudiDDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
+
+- pipeline = StableDiffusionPipeline.from_pretrained(
++ pipeline = GaudiStableDiffusionPipeline.from_pretrained(
+    model_name,
+    scheduler=scheduler,
++   use_habana=True,
++   use_lazy_mode=False,
```
Comment (libinta) [inline README.md:195]: remove this

File: docs/source/quickstart.mdx [hunk 2]
```diff
@@ -89,3 +89,38 @@ gaudi_config = GaudiConfig.from_pretrained(
...
+
+- scheduler = DDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
++ scheduler = GaudiDDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
+
+- pipeline = StableDiffusionPipeline.from_pretrained(
++ pipeline = GaudiStableDiffusionPipeline.from_pretrained(
+    model_name,
+    scheduler=scheduler,
++   use_habana=True,
++   use_lazy_mode=False,
```
Comment (libinta) [inline docs/source/quickstart.mdx:116]: remove

File: docs/source/usage_guides/stable_diffusion.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+
+## How to generate images?
+
+To generate images with Stable Diffusion on Gaudi, you need to instantiate two instances:
+- A pipeline with [`GaudiStableDiffusionPipeline`](https://huggingface.co/docs/optimum/habana/package_reference/stable_diffusion_pipeline). This pipeline supports *text-to-image generation*.
+- A scheduler with [`GaudiDDIMScheduler`](https://huggingface.co/docs/optimum/habana/package_reference/stable_diffusion_pipeline#optimum.habana.diffusers.GaudiDDIMScheduler). This scheduler has been optimized for Gaudi.
+
+When initializing the pipeline, you have to specify `use_habana=True` to deploy it on HPUs.
+Furthermore, in order to get the fastest possible generations you should enable **lazy mode** or **HPU graphs** with `use_lazy_mode=True` or `use_hpu_graphs=True`.
```
Comment (libinta) [inline docs/source/usage_guides/stable_diffusion.mdx:30]: change this portion to use HPU_GRAPH

File: docs/source/usage_guides/stable_diffusion.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+model_name = "CompVis/stable-diffusion-v1-4"
+
+scheduler = GaudiDDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
+
+pipeline = GaudiStableDiffusionPipeline.from_pretrained(
+    model_name,
+    scheduler=scheduler,
+    use_habana=True,
+    use_lazy_mode=False,
```
Comment (libinta) [inline docs/source/usage_guides/stable_diffusion.mdx:45]: remove

File: docs/source/usage_guides/stable_diffusion.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+model_name = "stabilityai/stable-diffusion-2-base"
+
+scheduler = GaudiDDIMScheduler.from_pretrained(model_name, subfolder="scheduler")
+
+pipeline = GaudiStableDiffusionPipeline.from_pretrained(
+    model_name,
+    scheduler=scheduler,
+    use_habana=True,
+    use_lazy_mode=False,
```
Comment (libinta) [inline docs/source/usage_guides/stable_diffusion.mdx:88]: remove

File: examples/stable-diffusion/text_to_image_generation.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    logger.setLevel(logging.INFO)
+
+    # Initialize the scheduler and the generation pipeline
+    scheduler = GaudiDDIMScheduler.from_pretrained(args.model_name_or_path, subfolder="scheduler")
+    pipeline = GaudiStableDiffusionPipeline.from_pretrained(
+        args.model_name_or_path,
+        scheduler=scheduler,
+        use_habana=args.use_habana,
+        use_lazy_mode=args.use_lazy_mode,
+        use_hpu_graphs=args.use_hpu_graphs,
```
Comment (libinta) [inline examples/stable-diffusion/text_to_image_generation.py:135]: remove

File: optimum/habana/diffusers/pipeline_utils.py [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        if self.use_lazy_mode and self.use_hpu_graphs:
+            raise ValueError("`use_lazy_mode` and `use_hpu_graphs` cannot be both True.")
+
+        if self.use_habana:
+            if self.use_lazy_mode:
+                logger.info("Enabled lazy mode.")
+            elif self.use_hpu_graphs:
+                logger.info("Enabled HPU graphs.")
+            else:
+                os.environ["PT_HPU_LAZY_MODE"] = "2"
```
Comment (libinta) [inline optimum/habana/diffusers/pipeline_utils.py:101]: remove lazy_mode as 2 since it's eager mode

Comment (regisss) [reply optimum/habana/diffusers/pipeline_utils.py:92]: Ah I may have misunderstood during the meeting yesterday: we want to remove eager mode and set lazy mode as the default right?

**Review by libinta — APPROVED** [review]

**Pull request merged**